### PR TITLE
[10.0][MIG] base_search_fuzzy

### PR DESCRIPTION
**Pull request by lasley**
Upgrade `base_search_fuzzy` to v10:
* Rename manifest
* Update openerp references to odoo
* Bump version
* Upgrade api usages

### CODE REVIEW

File: base_search_fuzzy/models/ir_model.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        table_alias = '"%s"' % (eleaf.generate_alias())
+
+        if operator == '%':
+
+            sql_operator = '%%'
+            params = []
+
+            if left in model._fields:
+                column = '%s.%s' % (table_alias, expression._quote(left))
+                query = '(%s %s %%s)' % (column, sql_operator)
```
Comment (lasley) [inline base_search_fuzzy/models/ir_model.py]: `%%s` should be replaced with [`model._fields[left].column_format`](https://github.com/odoo/odoo/blob/10.0/odoo/fields.py#L298) on this and ln 31

Comment (StefanRijnhart) [reply base_search_fuzzy/models/ir_model.py]: Did you review your own PR?

**Review by StefanRijnhart — APPROVED** [review]
Migration LGTM and tests run successfully. Thanks!

File: base_search_fuzzy/models/ir_model.py (hunk 1 above)
Comment (lasley) [reply base_search_fuzzy/models/ir_model.py]: Hah more of just a note to remember to make this change. I was randomly digging around in the expression code for an unrelated thing, saw `column_format`, and remembered this code. Making the change now!

**Comment by lasley** [comment]
@tedsalmon please review

File: base_search_fuzzy/README.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* add better `order by` handling
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/server-tools/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed feedback.
```
Comment (tedsalmon) [inline base_search_fuzzy/README.rst]: Old verbiage

File: base_search_fuzzy/models/ir_model.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# © 2016 Serpent Consulting Services Pvt. Ltd.
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import models, api
```
Comment (tedsalmon) [inline base_search_fuzzy/models/ir_model.py]: Alphabetize import order

File: base_search_fuzzy/models/ir_model.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                    sql_operator,
+                    model._fields[left].column_format,
+                )
+            elif left in models.MAGIC_COLUMNS:
+                query = "(%s.\"%s\" %s %%s)" % (
+                    table_alias, left, sql_operator)
+                params = right
+            else:  # Must not happen
+                raise ValueError(
+                    "Invalid field %r in domain term %r" % (left, leaf))
```
Comment (tedsalmon) [inline base_search_fuzzy/models/ir_model.py]: Exceptions should be internationalized, right?

**Review by tedsalmon — APPROVED** [review]
LGTM 👍 ! A few small review notes were left but they're likely things that existed pre-migration :)

**Comment by lasley** [comment]
Thanks @tedsalmon - comments attended to

**Pull request merged**